r-package/grf/R/boosted_regression_forest.R

```diff
@@ -0,0 +1,229 @@
+      step.error.approx <- predict(forest.small)$debiased.error
+      still_boosting <- (mean(step.error.approx,na.rm=TRUE) < tolerance*mean(error.debiased,na.rm=TRUE))
+    }
+
+    if(still_boosting) {
+      forest.resid <- regression_forest(X,Y.resid, sample.fraction = as.numeric(tunable.params["sample.fraction"]),
+                                  mtry = as.numeric(tunable.params["mtry"]), tune.parameters = FALSE,
+                                  num.trees = num.trees,
+                                  num.threads = num.threads,
+                                  min.node.size = as.numeric(tunable.params["min.node.size"]),
+                                  honesty = honesty,
+                                  honesty.fraction = honesty.fraction,
+                                  seed = seed, ci.group.size = ci.group.size,
+                                  alpha = as.numeric(tunable.params["alpha"]),
+                                  imbalance.penalty = as.numeric(tunable.params["imbalance.penalty"]),
+                                  clusters = clusters, samples_per_cluster = samples_per_cluster);
+
+      current.pred <- predict(forest.resid)
+      Y.hat <- Y.hat + current.pred$predictions
+      error.debiased <- current.pred$debiased.error
+      boosted.forest[["forests"]][[step]] <- forest.resid
+      boosted.forest[["debiased.errors"]][[step]]<- error.debiased
+      step<-step+1
+    }
+  }
+  boosted.forest[["predictions"]] <- Y.hat
+  class(boosted.forest) <- c("boosted_regression_forest")
+  boosted.forest
+}
+
+#' Predict with a boosted regression forest.
+#'
+#' Gets estimates of E[Y|X=x] using a trained regression forest.
+#'
+#' @param object The trained forest.
+#' @param newdata Points at which predictions should be made. If NULL, makes out-of-bag
+#'                predictions on the training set instead (i.e., provides predictions at
+#'                Xi using only trees that did not use the i-th training example). Note
+#'                that this matrix should have the number of columns as the training
+#'                matrix, and that the columns must appear in the same order.
+#' @param num.steps Number of boosting iterations to use for prediction. If blank, uses the full number of steps
+#'        for the object given
+#' @return A vector of predictions.
+#'
+#' @examples \dontrun{
+#' # Train a boosted regression forest.
+#' n = 50; p = 10
+#' X = matrix(rnorm(n*p), n, p)
+#' Y = X[,1] * rnorm(n)
+#' r.boosted.forest = boosted_regression_forest(X, Y)
+#'
+#' # Predict using the forest.
+#' X.test = matrix(0, 101, p)
+#' X.test[,1] = seq(-2, 2, length.out = 101)
+#' r.pred = predict(r.boosted.forest, X.test)
+#'
+#' # Predict on out-of-bag training samples.
+#' r.pred = predict(r.boosted.forest)
+#'
+#' }
+#'
+#' @method predict boosted_regression_forest
+#' @export
+predict.boosted_regression_forest <- function(object, newdata=NULL,
```

Comment:
We should make sure the `num.threads` parameter is available here as well. I agree that it makes sense to omit the other parameters like `estimate.variance`.